modules/jsonld/src/main/scala/ch/epfl/bluebrain/nexus/rdf/jsonld/JsonLd.scala

```diff
@@ -93,6 +104,183 @@ object JsonLd {
     )
   }
 
+  /**
+    * Adds or merges a context URI to an existing JSON object.
+    *
+    * @param json    the json
+    * @param context the standard context URI
+    * @return a new JSON object
+    */
+  def addContext(json: Json, context: AbsoluteIri): Json = {
+    val contextUriString = Json.fromString(context.asString)
+
+    json.asObject match {
+      case Some(jo) =>
+        val updated = jo("@context") match {
+          case None => jo.add("@context", contextUriString)
+          case Some(value) =>
+            (value.asObject, value.asArray, value.asString) match {
+              case (Some(vo), _, _) if vo.isEmpty =>
+                jo.add("@context", contextUriString)
+              case (_, Some(va), _) if va.isEmpty =>
+                jo.add("@context", contextUriString)
+              case (_, _, Some(vs)) if vs.isEmpty =>
+                jo.add("@context", contextUriString)
+              case (Some(vo), _, _) if !vo.values.exists(_ == contextUriString) =>
+                jo.add("@context", Json.arr(value, contextUriString))
+              case (_, Some(va), _) if !va.contains(contextUriString) =>
+                jo.add("@context", Json.fromValues(va :+ contextUriString))
+              case (_, _, Some(vs)) if vs != context.asString =>
+                jo.add("@context", Json.arr(value, contextUriString))
+              case _ => jo
+            }
+        }
+        Json.fromJsonObject(updated)
+      case None => json
+    }
+  }
+
+  /**
+    * @param json the primary context. E.g.: {"@context": {...}}
+    * @param that the other context to merge with this context. E.g.: {"@context": {...}}
+    * @return a new Json with the values of the ''@context'' key (this) and the provided ''that'' top ''@context'' key
+    *         If two keys inside both contexts collide, the one in the ''other'' context will override the one in this context
+    */
+  def mergeContext(json: Json, that: Json): Json =
+    Json.obj("@context" -> merge(contextValue(json), contextValue(that)))
+
+  /**
+    * @param json the primary context
+    * @param that the context to append to this json. E.g.: {"@context": {...}}
+    * @return a new Json with the original context plus the provided context
+    */
+  def appendContextOf(json: Json, that: Json): Json = json deepMerge mergeContext(json, that)
+
+  /**
+    * Filter out context which are strings/iris as Jena doesn't  handle them. Other invalid contexts(booleans, numbers) etc.
+    * will by handled by Jena and cause an error.
+    *
+    * @param json the json
+    * @return the context in the form {"@context": {...}}. The values that are not inside the key @context are dropped
+    */
+  def removeContextIris(json: Json): Json = {
```

Comment:
We should not need this method anymore, since:
1. We have a resolver mechanism to resolve iris to actual contexts
2. In a weird case that this will happen, Jena will not follow links in any case, since there is this environment variable (I forgot the name)